src/defs.hpp

```diff
@@ -56,21 +60,211 @@ struct LogicalLocation { // aggregate and POD type
   // These values can exceed the range of std::int32_t even if the root grid has only a
   // single MeshBlock if >30 levels of AMR are used, since the corresponding max index =
   // 1*2^31 > INT_MAX = 2^31 -1 for most 32-bit signed integer type impelementations
-  std::int64_t lx1, lx2, lx3;
-  int level;
+
+ private:
```

Comment:
`LogicalLocation` is now complex enough that maybe the struct deserves its own header file. Doesn't feel appropriate for it to live in `decs.h` anymore.

Also maybe it should be made a `class`. I know those are the same as structs, but now that it has private member data, I feel like the `class` designation better conveys intent.

Reply:
Agreed on both accounts

Reply:
done